Compiler/NFFrontEnd/NFInst.mo

```diff
@@ -732,9 +741,31 @@ function addInheritedElements2
   input ClassTree.Tree inScope;
   input Integer nodeIndex;
   output ClassTree.Tree scope;
+protected
+  SCode.Element c;
+  ClassTree.Entry idExisting;
+  InstNode nodeExisting;
 algorithm
   scope := match id
-    case ClassTree.Entry.CLASS() then ClassTree.add(inScope, name, id);
+    case ClassTree.Entry.CLASS()
+      algorithm
+        // check if class exists!
+        try
+          (idExisting as ClassTree.Entry.CLASS(nodeExisting)) := ClassTree.get(inScope, name);
```

Comment:
ClassTree.add (i.e. AvlTreeBase.add) takes a fourth argument which is a function called in the case of a collision. If you use that instead you don't incur the cost of a lookup for each class.